integrationtests/src/test/java/io/kroxylicious/proxy/KrpcFilterIT.java

```diff
@@ -149,6 +153,29 @@ public void shouldPassThroughRecordUnchanged(KafkaCluster cluster, Admin admin)
         }
     }
 
+    @Test
+    public void requestFiltersCanRespondWithoutProxying(KafkaCluster cluster, Admin admin) throws Exception {
+        String proxyAddress = "localhost:9192";
+
+        var config = baseConfigBuilder(proxyAddress, cluster.getBootstrapServers())
+                .addNewFilter().withType("CreateTopicRejectFilter").endFilter().build().toYaml();
+
+        try (var ignored = startProxy(config)) {
+            try (var proxyAdmin = Admin.create(Map.of(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, proxyAddress))) {
+                ExecutionException exception = assertThrows(ExecutionException.class, () -> {
+                    proxyAdmin.createTopics(List.of(new NewTopic(TOPIC_1, 1, (short) 1))).all().get();
+                });
+                Throwable cause = exception.getCause();
+                assertTrue(cause instanceof InvalidTopicException);
+                assertEquals(CreateTopicRejectFilter.ERROR_MESSAGE, cause.getMessage());
```

Comment:
Well that was an interesting diversion:

```java
Assertions.assertThatExceptionOfType(ExecutionException.class)
            .isThrownBy(() -> {
                proxyAdmin.createTopics(List.of(new NewTopic(TOPIC_1, 1, (short) 1))).all().get();
            })
            .withCauseInstanceOf(InvalidTopicException.class)
            .extracting(Throwable::getCause)
            .asInstanceOf(InstanceOfAssertFactories.THROWABLE)
            .hasMessage(CreateTopicRejectFilter.ERROR_MESSAGE)
```

I'm not sure that actually increases readability but was interesting to explore

Reply:
How about this:

```java
Assertions.assertThatExceptionOfType(ExecutionException.class)
                        .isThrownBy(() -> {
                            proxyAdmin.createTopics(List.of(new NewTopic(TOPIC_1, 1, (short) 1))).all().get();
                        })
                        .withCauseInstanceOf(InvalidTopicException.class)
                        .havingCause()
                        .withMessage(CreateTopicRejectFilter.ERROR_MESSAGE)
```

Reply:
Ohh yeah that works better 😁

Reply:
Pushed